### ISSUE TITLE
setting the *_DIR family of env variables does not override defaults

### ISSUE THREAD

**Issue by nega0** [PyTables/PyTables#1016]
Note that my perspective on this is that of a Homebrew user on Apple Silicon, who supports other Homebrew users on Apple Silico. Until PyTables supplies binary wheels, some variation of `pip install tables` needs to "just work".

One would expect that explicitly setting the location of dependencies would override defaults. This expectation is confirmed in the documentation at https://github.com/PyTables/PyTables/blob/577bd6662bacb0a989a6e32870d8df9c3724aacc/doc/source/usersguide/installation.rst?plain=1#L96-L105

Basically this only affects `blosc` and `blosc2` detection, as `hdf5`, `lzo`, and `bzip2` don't have pkg-config files. One _could_ set `USE_PKGCONFIG=FALSE`, but if one wanted both `blosc` and `blosc2` they would have to use some variation of

```
USE_PKGCONFIG=FALSE BLOSC_DIR=/opt/c-blosc BLOSC2_DIR=/opt/c-blosc2 python3 setup.py build
```

Currently it appears that dependency detection is
 1. `pkg-config`, then
 2.  paths in `("/usr/local", "/sw", "/opt", "/opt/local", "/usr", "/")`, then
 3.  the corresponding `*_DIR` env variable

(I didn't check the arguments to `setup.py` as I'm concerned with `pip`'ing PyTables in)

For the sake of argument, let's use `blosc2` since that's what's driving this. I expected something more like
 1. if `BLOSC2_DIR` exists, use its value. if `blosc2` can't be found there, bail on further detection. 
 2. the `--blosc2` argument to `setup.py`. again, bail on further detection if that fails
 3. `pkg-config`, if not found that way, then
 4. a selection of hardcoded paths

(I suppose 1 and 2 could be swapped, but that's my personal preference)

I feel like this would give the user the proper control and would enable a ready workaround for #998 until Blosc/python-blosc2#110 is resolved.

**Comment by bnavigator** [comment]
So you're saying the changes in #1000 have no effect and setting `BLOSC2_INCDIR`, `BLOSC2_LIBDIR` do not override the (incorrect) pkgconfig file?

Because here pkgconfig should only be called if those variables are not set:

https://github.com/PyTables/PyTables/blob/577bd6662bacb0a989a6e32870d8df9c3724aacc/setup.py#L103-L118

**Comment by nega0** [comment]
I never said anything about `BLOSC2_INCDIR` or `BLOSC2_LIBDIR`.

**Comment by avalentino** [comment]
@nega0, @bnavigator #1017 should hopefully fix the problem of setting the blosc2 path using CLI/ENV.
Could you please give it a try?

**Comment by avalentino** [comment]
Closed by #1017
